summarizer/datasets/correlation.ipynb

```diff
@@ -0,0 +1,432 @@
+     "execution_count": 16
+    }
+   ],
+   "source": [
+    "# Human scores: should be maximum\n",
+    "# Leave-one-out strategy\n",
+    "corrs = []\n",
+    "for i in range(n_users):\n",
+    "    machine_scores = user_scores[i]\n",
+    "    corrs.append(evaluate_scores(machine_scores, np.delete(user_summary, i, 0)))\n",
+    "np.mean(corrs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Using scores `[0.8 0.3 ... 0.1]` as target"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "execute_result",
+     "data": {
+      "text/plain": "0.0003529005001347934"
+     },
+     "metadata": {},
+     "execution_count": 17
+    }
+   ],
+   "source": [
+    "# Random scores: should be 0\n",
+    "corrs = []\n",
+    "for _ in range(50):\n",
+    "    machine_scores = np.random.uniform(0, 1, (n_frames,))\n",
+    "    corrs.append(evaluate_scores(machine_scores, user_scores))\n",
+    "np.mean(corrs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "execute_result",
+     "data": {
+      "text/plain": "0.2740630927883905"
+     },
+     "metadata": {},
+     "execution_count": 18
+    }
+   ],
+   "source": [
+    "# Human scores: should be maximum\n",
+    "# Leave-one-out strategy\n",
+    "corrs = []\n",
+    "for i in range(n_users):\n",
+    "    machine_scores = user_scores[i]\n",
+    "    corrs.append(evaluate_scores(machine_scores, np.delete(user_scores, i, 0)))\n",
+    "np.mean(corrs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "_Observations:_\n",
+    "* Using `user_summary` is consistent: random is near 0 and human is high (~0.17)\n",
+    "* Using `user_scores` matches the paper: random also near 0 and human close to 0.26 from paper (us ~0.27)\n",
+    "\n",
+    "In the paper they have 0.26 so they might have averaged over all videos in TVSum.\n",
+    "\n",
+    "We should use `user_scores` for correlation because `user_summary` is actually a summary, so KTS+Knapsack+15% constraint have been used. They do not reflect the absolute importance of a single frame."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# SumMe"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 89,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": ".h5:  b'Air_Force_One' (15, 4494) (1, 4494) 4494\n"
+    }
+   ],
+   "source": [
+    "dataset = h5py.File(\"summarizer_dataset_summe_google_pool5.h5\", \"r\")\n",
+    "video_1 = dataset[\"video_1\"]\n",
+    "video_name = video_1[\"video_name\"][...].item()\n",
+    "n_frames = video_1[\"n_frames\"][...]\n",
+    "positions = video_1[\"picks\"][...]\n",
+    "\n",
+    "user_summary = video_1[\"user_summary\"][...] # (n_users, n_frames)\n",
```

Comment:
We should re-add the line `n_users, n_frames = user_summary.shape` after this, or redefine `n_users` in some way, otherwise the second cell following this one uses the `n_users` value from the TVSum dataset. This causes an out-of-bounds error (there are more in TVSum than SumMe).

Reply:
Indeed thanks, fixed.